src/data_capture/video_capture/scripts/capture.py

```diff
@@ -9,6 +9,76 @@
 
 from video_recorder import VideoRecorder
 
+from threading import Timer
+import os
+from time import sleep
+
+import psutil
+
+
+"""
+This is a threaded timer to execute a piece of code periodically 
+without blocking the main execution.
+
+This non-blocking method allows the code to keep running while also
+executing our function every n seconds
+
+NOTE:
+- start() and stop() are safe to call multiple times even if the timer 
+has already started/stopped
+
+- You can change interval anytime, it will be effective after next run. 
+Same for args, kwargs and even function!
+"""
+
+class RepeatedTimer(object):
```

Comment:
This should probably be in its own file. 

Also, the comment above is typically right under the class definition. You can look at [pep257](https://www.python.org/dev/peps/pep-0257/) for best practices.